### PR TITLE
15: Team list page

### PR DESCRIPTION
**Pull request by meladawy**
1. Remove the token parameter from all services and components
2. Create a team list page 
3. Change the "Create a submission" button to point to the new teams listing page because submission is gonna be created from the teams page instead of the dashboard.

### CODE REVIEW

File: src/app/pages/user-team-page/user-team-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    this.userteamsEndpoint = await this.endpointService.find("userTeams");
+    this.initializeData();
+  }
+
+  initializeData() {
+    this.requestsService.get(this.userteamsEndpoint).subscribe(
+      data => {
+        try {
+          this.teams = data;
+        } catch (e) {}
```
Comment (ke4) [inline src/app/pages/user-team-page/user-team-page.component.ts]: Do we need this try-catch block here?

File: src/app/pages/user-team-page/user-team-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+   * Loop across the teams and get submissions for each team and put it in array.
+   */
+  getTeamsByUrl(serviceUrl) {
+    this.requestsService.get(serviceUrl).subscribe(
+      data => {
+        this.teams = data;
+      },
+      err => {
+        // TODO: Handle Errors.
```
Comment (ke4) [inline src/app/pages/user-team-page/user-team-page.component.ts]: Do we have a clear idea how to handle errors?

Comment (meladawy) [reply src/app/pages/user-team-page/user-team-page.component.ts]: We have a separate ticket for it...its gonna be still under discussion.

File: src/app/pages/user-team-page/user-team-page.component.ts (hunk 1 above)
Comment (meladawy) [reply src/app/pages/user-team-page/user-team-page.component.ts]: Ooh no, I forgot to remove it. Thanks for letting me know!

File: src/app/pages/user-team-page/user-team-page.component.ts (hunk 2 above)
Comment (ke4) [reply src/app/pages/user-team-page/user-team-page.component.ts]: OK

**Pull request merged**